### PR TITLE
Add DesignCreateMode field to the RTD

### PR DESCRIPTION
**Pull request by frutiemax**
Maze is a particular case when creating a TrackDesign. The solution is to add a DesignCreateMode field to the RTD, all the rides but the maze behave the same.

### CODE REVIEW

**Comment by spacek531** [comment]
Is there not already a maze flag in the rides flags?

**Comment by frutiemax** [comment]
I don't think there is a flag for a maze.

**Review by tupaschoal — COMMENTED** [review]
Same question about "defaulting"

File: src/openrct2/ride/RideData.h [hunk 1]
```diff
@@ -144,7 +150,6 @@ struct UpkeepCostsDescriptor
 };
 
 using RideTrackGroup = OpenRCT2::BitSet<TRACK_GROUP_COUNT>;
-
```
Comment (tupaschoal) [inline src/openrct2/ride/RideData.h:147]: Unrelated change, please revert

File: src/openrct2/ride/RideData.h [hunk 2]
```diff
@@ -77,6 +77,12 @@ enum class RideColourKey : uint8_t
     Toilets
 };
 
+enum class TrackDesignCreateMode : uint64_t
```
Comment (tupaschoal) [inline src/openrct2/ride/RideData.h:80]: Any reason for `uint64_t`? Unless it is to avoid replays breaking and stuff, it seems a bit of an overkill, could have been `bool`

Comment (Broxzier) [reply src/openrct2/ride/RideData.h:80]: This looks like a nice opportunity to use `int_fast8_t`.

**Comment by duncanspumpkin** [comment]
> I don't think there is a flag for a maze.

Yeah there isn't at present any flag. I think this is probably fine as we might in the future have more than 1 super special ride type that needs custom things like maze.

**Review by tupaschoal — APPROVED** [review]

**Comment by tupaschoal** [comment]
Please solve merge conflicts (probably best to do is to solve one, then wait for merge, then solve another)

**Review by Broxzier — APPROVED** [review]
